### ISSUE TITLE
Heartbeat event doesn't always trigger

### ISSUE THREAD

**Issue by jenniewerner** [transistorsoft/cordova-background-geolocation-lt#533]
Is the heartbeat event suppose to always trigger for iOS? I have the interval set to every 15 minutes, but if the app is the in the background for a long time, I stop seeing heartbeat events in my logs. Do things such as low-battery mode stop heartbeat triggers?

**Comment by christocracy** [comment]
When the device is unplugged and off, iOS freezes all timers.

Heartbeat will resume if you:
- shake or walk around with the device.
- unlock the device

**Comment by jenniewerner** [comment]
When I use a short heartbeat interval of a few minutes and have been recently using my app, I see the heartbeat logs no problem. But for example when I get home after my app has been in the background a log time, I don't see the heartbeat logs, even though I was recently walking and using my phone. Do you have any tips on how to debug this? It always work in test scenarios just not in real life scenarios.

**Comment by christocracy** [comment]
The plugin will hold 3 days of logs by default.  See [Debugging](https://github.com/transistorsoft/cordova-background-geolocation-lt/wiki/Debugging) to find out how to retrieve them.

```javascript
{
  logLevel: bgGeo.LOG_LEVEL_VERBOSE,
  debug: true
}
```

In your logs, you'll look for this:
```
🔵-[TSLocationManager onHeartbeat] Heartbeat
```

With `debug: true`, you'll hear the debug sound *boop* each time a `heartbeat` events occurs.  No *boop*, no `heartbeat`.

**Comment by jenniewerner** [comment]
Debugging with sounds isn't practical over a long period which is what I'm interested in. When I used notifications and turned off sound, I would stop getting debugging notifications after the app was left in the background for a long time (even if I was walking long distances). Once I opened the app, the notifications started up again.

I'm not sure how to get your logs with your functions. It looks like I need to have the code to get the logs run on the client when I'm setting up the package. How would I get logs retroactively?
